dojo/tools/scantist/parser.py

```diff
@@ -0,0 +1,82 @@
+import hashlib
+import json
+from dojo.models import Finding
+
+__author__ = 'mohcer'
+
+
+class ScantistJSONParser(object):
+    def __init__(self, file, test):
+        self.items = []
+
+        if file is None:
+            return
+
+        result_data = file.read()
+        try:
+            content = json.loads(str(result_data, 'utf-8'))
+        except:
+            content = json.loads(result_data)
+
+        if content is None:
+            return
+
+        self.items = [data for data in self.get_items(content, test)]
+
+    def get_items(self, tree, test):
+        """
+        tree list: input tree list of all the vulnerability findings
+        test:
+        : purpose: parses input rawto extract dojo
+        """
+        def get_findings(vuln, test):
+            """
+            vuln : input vulnerable node
+            test :
+            """
+            cve = vuln.get("Public ID")
+            component_name = vuln.get("Library")
+            component_version = vuln.get("Library Version")
+
+            title = cve + '|' + component_name
+            description = vuln.get("Description")
+
+            file_path = vuln.get("File Path", "")
+            sourcefile = None
+
+            severity = vuln.get("Score", "N/A")
+            score = vuln.get("cvss_v3_score", "N/A")
+
+            mitigation = vuln.get("Patched Version")
+
+            finding = Finding(
+                title=title,
+                test=test,
+                description=description,
+                severity=severity,
```

Comment:
There will be a pb with this one. If you don't have data for severity, the value "N/A" will produce exceptions when saving the Finding.

Reply:
From the look of it 
```
        if severity == 'Critical':
            return 'S0'
        elif severity == 'High':
            return 'S1'
        elif severity == 'Medium':
            return 'S2'
        elif severity == 'Low':
            return 'S3'
        elif severity == 'Info':
            return 'S4'
        else:
            return 'S5'
 ```
 
 should it return S5? is S5 unknown severity something of this sort?